insights/parsers/ls_run_systemd_generator.py

```diff
@@ -0,0 +1,46 @@
+"""
+LsRunSystemdGenerator - command ``ls -lan /run/systemd/generator``
+==================================================================
+
+The ``ls -lan /run/systemd/generator`` command provides information for only
+the ``/run/systemd/generator`` directory.
+
+Sample input is shown in the Examples. See ``FileListing`` class for
+additional information.
+
+Sample directory list::
+
+    total 28
+    drwxr-xr-x.  6 0 0 260 Aug  5 07:35 .
+    drwxr-xr-x. 18 0 0 440 Aug  5 07:35 ..
+    -rw-r--r--.  1 0 0 254 Aug  5 07:35 boot.mount
+    -rw-r--r--.  1 0 0 259 Aug  5 07:35 boot\x2dfake.mount
+    -rw-r--r--.  1 0 0 176 Aug  5 07:35 dev-mapper-rhel\x2dswap.swap
+    drwxr-xr-x.  2 0 0 100 Aug  5 07:35 local-fs.target.requires
+    -rw-r--r--.  1 0 0 217 Aug  5 07:35 -.mount
+    drwxr-xr-x.  2 0 0  60 Aug  5 07:35 nfs-server.service.d
+    drwxr-xr-x.  2 0 0 100 Aug  5 07:35 remote-fs.target.requires
+    -rw-r--r--.  1 0 0 261 Aug  5 07:35 root-mnt_nfs3.mount
+    -rw-r--r--.  1 0 0 261 Aug  5 07:35 root-mnt\x2dnfs1.mount
+    -rw-r--r--.  1 0 0 261 Aug  5 07:35 root-mnt\x2dnfs2.mount
+    drwxr-xr-x.  2 0 0  60 Aug  5 07:35 swap.target.requires
+
+Examples:
+
+    >>> ls.files_of(None) == ['boot.mount', 'boot-fake.mount', 'dev-mapper-rhel-swap.swap', '-.mount', 'root-mnt_nfs3.mount', 'root-mnt-nfs1.mount', 'root-mnt-nfs2.mount']
```

Comment:
Use `None` as a parameter is ambiguous, the base `FileListing` have resolved this issue by parsing the `file_path`. But it needs a valid argument being passed.

See: https://github.com/RedHatInsights/insights-core/blob/9d277560c6a725266f48993e25d1c3d6b7b7ba41/insights/parsers/tests/test_ls_var_lib_mongodb.py#L18

Reply:
The real path is used instead of None.